static/js/compose_actions.js

```diff
@@ -292,10 +306,32 @@ export function start(msg_type, opts) {
         return;
     }
     notifications.clear_compose_notifications();
-    expand_compose_box();
 
     opts = fill_in_opts_from_current_narrowed_view(msg_type, opts);
 
+    // Close and block compose box for the pm-with case, if
+    // the recipient is a deactivated user else,
+    // expand compose box.
+    if (
+        opts.private_message_recipient !== "" &&
+        check_pm_deactivated(
+            people.emails_strings_to_user_ids_string(opts.private_message_recipient),
+        ).is_user_id_deactivated
+    ) {
+        // Expand compose box with a deactivated user also,
+        // if it is the case with stream or new private message button.
+        if (
+            (msg_type === "stream" && opts.trigger === "new topic button") ||
+            (msg_type === "private" && opts.trigger === "new private message")
```

Comment:
Let's combine these two `if` conditions. Maybe add a `check_new_top_or_pm_trigger()` to simplify reading it.

Also, we should update the compose box deactivated state once a user is deactivated or activated back.

Reply:
Updated the PR with what I was thinking here.

Reply:
Thanks.